### PR TITLE
Use PyPi version of BSyncPy and Update to BuildingSync 2.4

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
The Jupyter notebook example was on version 2.3.

#### What does this PR do?
* Update to 2.4
* Update python dependencies in poetry.lock

#### How should this be manually tested?
* Run the jupyter notebook

#### What are the relevant tickets?
#411

### CODE REVIEW

**Comment by nllong** [comment]
DO NOT MERGE until pointing the poetry file to a pypi release of this package.

File: docs/notebooks/bsync_examples/example1.xml
Comment (nllong) [inline docs/notebooks/bsync_examples/example1.xml:2]: after fixing and running the entire notebook, this was the only change to the example file, which means that all is still right with the world.

File: docs/notebooks/bsync_examples/Small-Office-Level-1.ipynb
Comment (nllong) [inline docs/notebooks/bsync_examples/Small-Office-Level-1.ipynb:2]: There are many changes to this file, easiest to view in Code/Browser.

File: docs/notebooks/bsync_examples/example1.xml
Comment (macintoshpie) [reply docs/notebooks/bsync_examples/example1.xml:2]: phew

**Review by macintoshpie — APPROVED** [review]
Just ran through the notebook in the browser and it all looks good

**Review by nllong — COMMENTED** [review]
@macintoshpie  -- can you check this again before I merge?

File: docs/notebooks/pyproject.toml
Comment (macintoshpie) [inline docs/notebooks/pyproject.toml:10]: 😎

**Review by macintoshpie — APPROVED** [review]
LGTM!

**Pull request merged**
